Bug in Threadpool test

Original report by Manuel:

```
../tests/test_ThreadPool.cc: In function ‘void CATCH2_INTERNAL_TEST_40()’:
../tests/test_ThreadPool.cc:588:10: error: ‘sort’ is not a member of ‘std’;
did you mean ‘qsort’?
  588 |     std::sort(output.begin(), output.end());
      |          ^~~~
      |          qsort
```

Including `<algorithm>` in `test_ThreadPool.cc` fixes that.

--- a/tests/test_ThreadPool.cc
+++ b/tests/test_ThreadPool.cc
@@ -22,6 +22,7 @@
 
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
+#include <algorithm>
 #include <array>
 #include <atomic>
 #include <stdexcept>

